### PR TITLE
Implement Go's varint 64 bit integer decoding in binary decoder

### PR DESCRIPTION
**Pull request by ddelnano**
Summary: Implement Go's varint 64 bit integer decoding in binary decoder

Our go version detection has been the source of a few bugs (#1300, #1318). This has highlighted that our go version detection is very different than how the `go version` cli works. In order to align our version detection with the go cli, we need to be able to decode these varints since parts of Go's buildinfo header use this encoding ([source](https://github.com/golang/go/blob/1dbbafc70fd3e2c284469ab3e0936c1bb56129f6/src/debug/buildinfo/buildinfo.go#L185-L190)).

Relevant Issues: #1318 #1300

Type of change: /kind feature

Test Plan: Used the following [go program](https://go.dev/play/p/x4x48NcxBMC) to draft the new test cases

Screenshot 2023-06-28 at 8 51 11 AM

### CODE REVIEW

**Review by vihangm — COMMENTED** [review]
Seems to line up with the go impl at https://github.com/golang/go/blob/master/src/encoding/binary/varint.go#L62-L88

Do you want to consider adding some more edge case tests like https://github.com/golang/go/blob/master/src/encoding/binary/varint_test.go#L163-L180 ?

File: src/stirling/utils/binary_decoder.h [hunk 1]
```diff
@@ -60,6 +62,32 @@ class BinaryDecoder {
...
 
+  // Extract UVarInt encoded value and return result as uint64_t. The details of this encoding's
+  // specification can be see in the following link:
+  // https://cs.opensource.google/go/go/+/refs/tags/go1.20.5:src/encoding/binary/varint.go;l=7-25
+  StatusOr<uint64_t> ExtractUVarInt() {
+    uint64_t x = 0;
+    uint bits = 0;
+    int i = 0;
+
+    while (buf_.size() >= 1 && i < kMaxVarintLen64) {
```
[the remark this answers is not in the document]
Comment (ddelnano) [reply src/stirling/utils/binary_decoder.h:73]: Updated in latest change.

**Review by vihangm — APPROVED** [review]

**Review by etep — APPROVED** [review]
